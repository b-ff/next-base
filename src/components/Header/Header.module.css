.header {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: var(--spacing_4x) var(--spacing_6x);
}
