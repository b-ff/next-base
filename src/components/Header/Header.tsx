"use client";
import { FC, ReactNode } from "react";
import styles from "./Header.module.css";
import { Auth } from "../Auth";
import Link from "next/link";
import { Flex } from "../Flex";

export type HeaderProps = {
  children?: ReactNode;
};

/**
 * Header of the app
 */
export const Header: FC<HeaderProps> = ({ children }) => {
  return (
    <header className={styles.header}>
      <Flex direction="row" gap={2}>
        <Link href="/">Home</Link> &raquo;
      </Flex>
      {children}
      <Auth />
    </header>
  );
};
