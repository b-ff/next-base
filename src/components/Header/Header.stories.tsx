import type { Meta, StoryObj } from "@storybook/react";

import { Header } from "./Header";

const meta: Meta<typeof Header> = {
  component: Header,
  tags: ["autodocs"],
};

export default meta;

type Story = StoryObj<typeof Header>;

export const Primary: Story = {
  args: {
    children: "Content",
  },
};
