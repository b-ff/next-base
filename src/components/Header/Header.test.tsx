import { Header } from "./Header";
import { render } from "@testing-library/react";

const mockChildren = "Content";

describe("Component Header", () => {
  it("Should render as expected", () => {
    const { container } = render(<Header>{mockChildren}</Header>);
    expect(container).toMatchSnapshot();
  });
});
