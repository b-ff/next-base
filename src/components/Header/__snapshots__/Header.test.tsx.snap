// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Header Should render as expected 1`] = `
<div>
  <header
    class="header"
  >
    <section
      style="display: flex; flex-direction: row; gap: var(--spacing_2x);"
    >
      <div>
        <a
          href="/"
        >
          Home
        </a>
      </div>
      <div>
         Â»
      </div>
    </section>
    Content
    <a
      href="#"
    >
      Sign In / Sign Up
    </a>
  </header>
</div>
`;
